### ISSUE TITLE
Feature request: support DateTimeOffset data type

### ISSUE THREAD

**Issue by ngoquoc** [Rhetos/Rhetos#53]
After several projects dealing with time zone differences, which resulted in quite tricky and fragile solutions, I think that Rhetos itself should support **DateTimeOffset** data type. There are some good reasons to have **DateTimeOffset** supported:
1. Database should be portable, so it can be restored on any server without having to care about server's timezone. Instead, timezone differences should be handled by the database engine by reading the timezone offset information stored in DateTimeOffset values.
2. There are so many use cases that need precise timestamps: accounting, logging, ... This need obviously can not be handled by the database server nor the application layer. Of course we can make an agreement that every timestamp value in the database is in a specific timezone (let's say UTC), but this approach requires a lot of unnecessary conversions from one timezone to the other. Also, I think handling timezone conversions in application layer is very error-prone, which we should avoid as much as possible.

**Comment by sstublic** [comment]
I'm kind of biased that DateTimeOffset in a database is an anti-pattern. I've used it once and will never again :) I've encountered many problems regarding queries and translation to different timezones.

Servers that are expected to deal with multiple timezones should always have an UTC time - which is the standard for all cloud providers. Having server in UTC and using DateTime provides unambiguous time point which you can then present to user and other service any way you like.

Not trying to invalidate this request, maybe there are specific situations where this is warranted. Just sharing a very bad experience.

**Comment by ngoquoc** [comment]
I agree that UTC time is the standard one to use, but it requires a lot of attention of developers to use it right. I met so many developers who were really confused when seeing a DateTime value, since they don't know if the value is UTC one, or a local one, unless it is explicitly named: UtcSomething. The problem is, not every developer pay that much attention to name things properly :)

I think DatetimeOffset would make developer's life easier since it is portable, it has all of the necessary information so you can easily know the correct time value by just looking at its properties, and most importantly, you don't have to guess which timezone it is.

About the compatibility, I believe that most of the vendors/libraries/software pieces have support of DateTimeOffset. It can be represented in [ISO-8601 format](https://www.w3.org/TR/NOTE-datetime), which I think a standard format to be communicated between systems.
